### PR TITLE
Relax error handling in transport disk usage API

### PR DESCRIPTION
**Pull request by dnhatn**
Relax the failure handling to cover other exceptions such as BroadcastShardOperationFailedException.

An assertion was tripped while @nik9000 was integrating the disk usage API to Rally.

```
» [2022-03-03T13:15:00,065][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [runTask-0] fatal error in thread [elasticsearch[runTask-0][analyze][T#1]], exiting
»  java.lang.AssertionError: unknown response [[logs-191998/AM-V3iMAS4W1CceHvQtajg][[logs-191998][2]] org.elasticsearch.action.support.broadcast.BroadcastShardOperationFailedException:]
»       at org.elasticsearch.action.admin.indices.diskusage.TransportAnalyzeIndexDiskUsageAction.newResponse(TransportAnalyzeIndexDiskUsageAction.java:115) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.action.admin.indices.diskusage.TransportAnalyzeIndexDiskUsageAction.newResponse(TransportAnalyzeIndexDiskUsageAction.java:42) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.action.support.broadcast.TransportBroadcastAction$AsyncBroadcastAction.finishHim(TransportBroadcastAction.java:258) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.action.support.broadcast.TransportBroadcastAction$AsyncBroadcastAction.onOperation(TransportBroadcastAction.java:206) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.action.support.broadcast.TransportBroadcastAction$AsyncBroadcastAction$1.handleResponse(TransportBroadcastAction.java:186) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.action.support.broadcast.TransportBroadcastAction$AsyncBroadcastAction$1.handleResponse(TransportBroadcastAction.java:178) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.transport.TransportService$4.handleResponse(TransportService.java:718) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1339) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»       at org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1417) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
```

### CODE REVIEW

**Comment by dnhatn** [comment]
run elasticsearch-ci/part-2

File: server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/TransportAnalyzeIndexDiskUsageAction.java [hunk 1]
```diff
@@ -89,9 +90,12 @@ protected AnalyzeDiskUsageShardResponse shardOperation(AnalyzeDiskUsageShardRequ
         final CancellableTask cancellableTask = (CancellableTask) task;
         final Runnable checkForCancellation = cancellableTask::ensureNotCancelled;
         final IndexShard shard = indicesService.indexServiceSafe(shardId.getIndex()).getShard(shardId.id());
+        shard.store().incRef();
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/TransportAnalyzeIndexDiskUsageAction.java:93]: Should we move this to the InternalEngine instead?

Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/TransportAnalyzeIndexDiskUsageAction.java]: I've opened https://github.com/elastic/elasticsearch/pull/84776 for this.

**Review by nik9000 — COMMENTED** [review]
I can't really comment on the code because I don't know the area but it does seem to fix my problem somewhat.

Now I'm getting rejected execution exceptions, but that's better!

**Review by nik9000 — APPROVED** [review]
Now I'm pretty sure I can approve this one!

**Comment by dnhatn** [comment]
Thanks Nik!

**Pull request merged**